common/djangoapps/student/cookies.py

```diff
@@ -0,0 +1,130 @@
+"""
+Utility functions for setting "logged in" cookies used by subdomains.
+"""
+
+import time
+import json
+
+from django.utils.http import cookie_date
+from django.conf import settings
+from django.core.urlresolvers import reverse, NoReverseMatch
+
+
+def set_logged_in_cookies(request, response, user):
+    """
+    Set cookies indicating that the user is logged in.
+
+    Some installations have an external marketing site configured
+    that displays a different UI when the user is logged in
+    (e.g. a link to the student dashboard instead of to the login page)
+
+    Currently, two cookies are set:
+
+    * EDXMKTG_LOGGED_IN_COOKIE_NAME: Set to 'true' if the user is logged in.
+    * EDXMKTG_USER_INFO_COOKIE_VERSION: JSON-encoded dictionary with user information (see below).
+
+    The user info cookie has the following format:
+    {
+        "version": 1,
+        "username": "test-user",
+        "email": "test-user@example.com",
+        "header_urls": {
+            "account_settings": "https://example.com/account/settings",
+            "learner_profile": "https://example.com/u/test-user",
+            "logout": "https://example.com/logout"
+        }
+    }
+
+    Arguments:
+        request (HttpRequest): The request to the view, used to calculate
+            the cookie's expiration date based on the session expiration date.
+        response (HttpResponse): The response on which the cookie will be set.
+        user (User): The currently logged in user.
+
+    Returns:
+        HttpResponse
+
+    """
+    if request.session.get_expire_at_browser_close():
+        max_age = None
+        expires = None
+    else:
+        max_age = request.session.get_expiry_age()
+        expires_time = time.time() + max_age
+        expires = cookie_date(expires_time)
+
+    cookie_settings = {
+        'max_age': max_age,
+        'expires': expires,
+        'domain': settings.SESSION_COOKIE_DOMAIN,
+        'path': '/',
+        'secure': None,
```

Comment:
Looking at this more closely, I'm inclined to set `secure` to "True" to ensure that these cookies are transmitted only with TLS.  This might make testing more difficult, but I think it's necessary if we're including the email and username in the cookie.